File: services/website/Gigmatch.Website.Model/Vacancy.cs
namespace Gigmatch.Website.Model;

public class Vacancy
{
    public class ExperienceItem
    {
        public Skill Skill { get; set; }

        public int Level { get; set; }
    }

    public Guid Id { get; init; }

    public string JobTitle { get; set; }

    public List<ExperienceItem> Experience { get; set; }

    public DateTime Start { get; set; }

    public DateTime End { get; set; }

    public DateTime Deadline { get; set; }
}

File: services/website/shared/src/main/java/nl/runnable/gigmatch/model/Vacancy.java
package nl.runnable.gigmatch.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Collections;
import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Vacancy {

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class ExperienceItem {
        private Skill skill;

        private int level;
    }

    private UUID id;

    private String jobTitle;

    private List<ExperienceItem> experience = Collections.emptyList();

}
